core/sail/fts/lucene-api/src/main/java/org/eclipse/rdf4j/sail/lucene/AbstractReaderMonitor.java

```diff
@@ -79,10 +88,10 @@ public boolean closeWhenPossible()
 		return closed.get();
 	}
 
-	public void close()
+	public final void close()
 		throws IOException
 	{
-		if (!closed.getAndSet(true)) {
+		if (closed.compareAndSet(false, true)) {
```

Comment:
This line is an actual bug fix, as getAndSet doesn't have a similar contract to the useful compareAndSet method, but on its own it didn't fix the issue.